### PR TITLE
Fixes #148  -- Document deploying on Kubernetes

### LINKED ISSUE #9: As sysadmin, I want to know how to deploy OpenZaak on kubernetes
... so I can do this at my municipality.

**Description**
There should be a document that describes what needs to be done to do this.

### LINKED ISSUE #148: As developer, I want to set up a test kubernetes for OpenZaak
... so I can do performance tests and create/verify documentation to setup such an environment.

**Description**
We need to set up a temporary kubernetes cluster to do both performance testing, find proper setups, and document our findings.

### PR DESCRIPTION
**Pull request by sergei-maertens**
Fixes #9 as well.

I'm sure there is a lot to improve on :-)

### CODE REVIEW

File: docs/installation/deployment/kubernetes.rst [hunk 1]
```diff
@@ -0,0 +1,280 @@
...
+To deploy Open Zaak, some variables need to be set (in ``vars/opennotificaties.yml``):
+
+* ``opennotificaties``: the domain name, e.g. ``notificaties.gemeente.nl``
+* ``opennotificaties_secret_key``: generate a key via https://miniwebtool.com/django-secret-key-generator/.
+  Make sure to put the value between single quotes!
+
+See ``roles/opennotificaties/defaults/main.yml`` for other possible variables to
+override.
+
+Deplying the applications
```
Comment (joeribekker) [inline docs/installation/deployment/kubernetes.rst]: Deploying*

File: docs/installation/deployment/kubernetes.rst [hunk 2]
```diff
@@ -0,0 +1,280 @@
...
+* ``openzaak_secret_key``: generate a key via https://miniwebtool.com/django-secret-key-generator/.
+  Make sure to put the value between single quotes!
+
+See ``roles/openzaak/defaults/main.yml`` for other possible variables to
+override.
+
+Configuring Open Notificaties
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+To deploy Open Zaak, some variables need to be set (in ``vars/opennotificaties.yml``):
```
Comment (joeribekker) [inline docs/installation/deployment/kubernetes.rst]: Open Notificaties*

File: docs/installation/deployment/kubernetes.rst [hunk 3]
```diff
@@ -0,0 +1,280 @@
...
+Application architecture
+========================
+
+Typically, your K8s cluster provider will set up a load balancer in front
+of your cluster, directing all the traffic to services in the cluster.
+
+DNS is set up so that your domain name(s) point to the IP address of the load
+balancer.
+
+The load balancer will then pass the traffic to an *Ingress* in the cluster,
```
Comment (joeribekker) [inline docs/installation/deployment/kubernetes.rst]: Add link to Ingress

File: docs/installation/deployment/kubernetes.rst [hunk 4]
```diff
@@ -0,0 +1,280 @@
...
+which will route the traffic to the backend services.
+
+All applications make use of caching via Redis cache databases.
+
```
Comment (joeribekker) [inline docs/installation/deployment/kubernetes.rst]: An image with an overview will be helpful.

File: docs/installation/deployment/kubernetes.rst [hunk 5]
```diff
@@ -0,0 +1,280 @@
...
+Kubernetes (K8s) is one of the main supported targets for deployment. We
+provide a fully automated deployment toolset suitable for DevOps engineers
+and/or system administrators.
+
+This documentation will sketch out the architecture, document the initial
+requirements and teach you how to deploy on K8s.
+
+This setup is tested against a Google Cloud Kubernetes cluster with 4vCPU and
+15G of memory.
```
Comment (joeribekker) [inline docs/installation/deployment/kubernetes.rst]: Link to n2 platform specs and possibly performance.

File: docs/installation/deployment/kubernetes.rst [hunk 6]
```diff
@@ -0,0 +1,280 @@
...
+    * the username of a superuser (typically ``postgres``)
+    * a password for the superuser
+    * credentials for the Open Zaak database and Open Notificaties database
+
+* A persistent-volume storage class supporting ``ReadWriteMany``. Contact your
+  provider to see if they offer it. If this is not an option, you can use a
+  ``ReadWriteOnce`` storage class and set up an NFS-server around it, but this
+  will likely have slower performance.
+
+  On Google Cloud, you can use:
```
Comment (joeribekker) [inline docs/installation/deployment/kubernetes.rst]: Perhaps full gcloud command list to setup the needed requirements for production.

File: docs/installation/deployment/kubernetes.rst [hunk 7]
```diff
@@ -0,0 +1,280 @@
...
+able to do this, you need superuser access. See the
+``vars/db_credentials.example.yml`` file for the example configuration.
+
+Both Open Zaak and Open Notificaties require database configuration to be
+defined in the ``vars/openzaak.yml`` and ``vars/opennotificaties.yml``
+variable files:
+
+.. code-block:: yaml
+
+    openzaak_db_name: openzaak  # name of the database to create
```
Comment (joeribekker) [inline docs/installation/deployment/kubernetes.rst]: Should we convert this to be a connect url?

File: docs/installation/deployment/kubernetes.rst [hunk 8]
```diff
@@ -0,0 +1,280 @@
+=======================
+Deploying on Kubernetes
+=======================
+
```
Comment (joeribekker) [inline docs/installation/deployment/kubernetes.rst]: Let's indicate the knowledge level of k8s to deploy this. And add some links to external resources to get familiar with our Target platform (gcloud).

File: docs/installation/deployment/kubernetes.rst (hunk 4 above)
Comment (sergei-maertens) [reply docs/installation/deployment/kubernetes.rst]: Yeah that's going to be a bit of work, to be continued :)

File: docs/installation/deployment/kubernetes.rst (hunk 7 above)
Comment (sergei-maertens) [reply docs/installation/deployment/kubernetes.rst]: No, it messes with the Ansible parameters and the Django envvars

File: docs/installation/deployment/kubernetes.rst (hunk 8 above)
Comment (sergei-maertens) [reply docs/installation/deployment/kubernetes.rst]: Added a section :+1:

File: docs/installation/deployment/kubernetes.rst (hunk 5 above)
Comment (sergei-maertens) [reply docs/installation/deployment/kubernetes.rst]: Added

File: docs/installation/deployment/kubernetes.rst (hunk 3 above)
Comment (sergei-maertens) [reply docs/installation/deployment/kubernetes.rst]: Mentioned the concept in the intro paragraph now with a link to the k8s docs

File: docs/installation/deployment/kubernetes.rst (hunk 4 above)
Comment (sergei-maertens) [reply docs/installation/deployment/kubernetes.rst]: Created #244 for this

File: docs/installation/deployment/kubernetes.rst (hunk 6 above)
Comment (sergei-maertens) [reply docs/installation/deployment/kubernetes.rst]: Added those to the introduction/requirements section

**Pull request merged**